### DISCUSSION TITLE
why npm is not support for deno?

### DISCUSSION THREAD

**Discussion by sunw31** [denoland/deno#15001] [Q&A]
let us do a assume... 

npm cli can know current workspace  is deno project or nodejs project
```
npm i xxx
```
npm cli can auto install deno module or nodejs module

so

every developer can publish their module in npm easily support nodejs or deno

npm ecosystem is really big importance.

**Comment by ghost** [comment]
This was covered in https://github.com/denoland/deno/discussions/10566#discussioncomment-2448134:
> npm is owned by GitHub which is owned by Microsoft which is a publicly traded company. It is more appropriate to say that the npm registry is a closed platform with closed governance. It has been a desire by Deno to not force users to lock-in to a single closed ecosystem.

**Comment by Abiti-936** [comment]
for me supporting Npm in Deno isn't a concern. If a package that's in Npm can run in Deno, you can link the package from Skypack or esm.sh that's what I do when I use Firebase JS SDK, Notion Client & GraphQL Yoga.

https://esm.sh/
https://www.skypack.dev/blog/2021/02/skypack-npm-packages-in-deno/
